Fix/switch to newer ur ros driver repo

jrgnicho opened the pull request:
This PR updates the pick and place demo to use the more recent universal ros driver repository https://github.com/UniversalRobots/Universal_Robots_ROS_Driver.
 This has not been tested on the robot yet

jrgnicho committed:
updated xacro to comply with current universal robot macro xacro

jrgnicho committed:
resolved build issues in noetic

jrgnicho committed:
updated config files to match changes in robot workcell urdf

jrgnicho committed:
updated list of source dependencies, not tested on robot yet

jdlangs commented on exercises/Perception-Driven_Manipulation/solution_ws/src/collision_avoidance_pick_and_place/src/tasks/pickup_box.cpp:
```diff
@@ -31,7 +31,6 @@ void collision_avoidance_pick_and_place::PickAndPlace::pickup_box(std::vector<ge
      * - The WRIST_LINK_NAME field in the "cfg" configuration member contains
      *  the name for the arm's wrist link.
      */
-    move_group_ptr->setEndEffector(cfg.WRIST_LINK_NAME);
```
Should this have been deleted?

jrgnicho replied:
It was not necessary since the manipulator group contains the tcp link

jdlangs reviewed (commented):
Seems fine overall to me. Shouldn't there also be similar changes in the `template_ws/` as well though?

jrgnicho commented:
> Seems fine overall to me. Shouldn't there also be similar changes in the `template_ws/` as well though?

Yes, I'll add them

jrgnicho committed:
copied changes into template_ws

jrgnicho committed:
updated pick and place moveit config package

jrgnicho committed:
removed unused launch file

jrgnicho committed:
updated pick and place application launch files

jrgnicho committed:
copied updated moveit config package into template directory

jrgnicho committed:
copied updated pick and place application launch files

jrgnicho commented:
@jdlangs I've updated the entire moveit_config package for the pick and place demo with the moveit setup assistant.  I also added the ros_control and ur stuff to allow running on the real robot.  I verified that it runs on it as well.  The template directory has been updated as well.
I'd suggest merging this before #348 so that we can then copy this updated version of the pick and place demo into the ros1 directory
